streams/src/main/java/org/apache/kafka/streams/state/internals/WindowKeySchema.java

```diff
@@ -49,7 +43,21 @@ public long segmentTimestamp(final Bytes key) {
 
     @Override
     public HasNextCondition hasNextCondition(final Bytes binaryKey, final long from, final long to) {
-        return ITERATOR_HAS_NEXT;
+        return new HasNextCondition() {
+            @Override
+            public boolean hasNext(final KeyValueIterator<Bytes, ?> iterator) {
+                if (iterator.hasNext()) {
+                    final Bytes bytes = iterator.peekNextKey();
+                    final Bytes keyBytes = WindowStoreUtils.bytesKeyFromBinaryKey(bytes.get());
+                    if (!keyBytes.equals(binaryKey)) {
```

Comment:
@xvrl Is this the same as this unit test we added here? https://github.com/apache/kafka/pull/2662/files#diff-b28e9353a9574fb2b6a33cfa6e560857R200